FirebaseInstallations/Source/Library/IIDMigration/FIRInstallationsIIDStore.m

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FIRInstallationsIIDStore.h"
+
+#if __has_include(<FBLPromises/FBLPromises.h>)
+#import <FBLPromises/FBLPromises.h>
+#else
+#import "FBLPromises.h"
+#endif
+
+#import <CommonCrypto/CommonDigest.h>
+#import "FIRInstallationsErrorUtil.h"
+
+static NSString *const kFIRInstallationsIIDKeyPairPublicTagPrefix =
+    @"com.google.iid.keypair.public-";
+static NSString *const kFIRInstallationsIIDKeyPairPrivateTagPrefix =
+    @"com.google.iid.keypair.private-";
+static NSString *const kFIRInstallationsIIDCreationTimePlistKey = @"|S|cre";
+
+@implementation FIRInstallationsIIDStore
+
+- (FBLPromise<NSString *> *)existingIID {
+  return [FBLPromise onQueue:dispatch_get_global_queue(QOS_CLASS_UTILITY, 0)
+                          do:^id _Nullable {
+                            if (![self hasPlistIIDFlag]) {
+                              return nil;
+                            }
+
+                            NSData *IIDPublicKeyData = [self IIDPublicKeyData];
+                            return [self IIDWithPublicKeyData:IIDPublicKeyData];
+                          }]
+      .validate(^BOOL(NSString *_Nullable IID) {
+        return IID.length > 0;
+      });
+}
+
+- (FBLPromise<NSNull *> *)deleteExistingIID {
```

Comment:
Do we also need to delete IID Token?
I wonder if you just call IID deleteID method which deletes everything. Although you still need to make sure to not let IID auto generate a new IID/Token when autoInitEnabled flag is on. (autoInitEnabled is on, iid regenerates a new iid/token but we can add work around)

Reply:
As per discussion with Rayo, we will need to remove IID when FIS deletes installation, so the old IID is not migrated again. If the developer decides to revert back to the version of IID that does not use FIS to generate the identifier, it will be expected the new Instance ID to be generated. So, I think, we will be fine here once I add deleting IID on deleting the FIS installation.